src/visitors/main.cpp

```diff
@@ -41,6 +42,26 @@ using namespace fmt::literals;
  * \brief Standalone program demonstrating usage of different visitors and driver classes.
  **/
 
+template <typename T>
+struct ClassInfo {
+    std::shared_ptr<T> v;
+    std::string id;
+    std::string description;
+};
+using VisitorInfo = ClassInfo<Visitor>;
+using ConstVisitorInfo = ClassInfo<ConstVisitor>;
+
+template <typename Visitor>
+void visit_program(const std::string& mod_file,
+                   const ClassInfo<Visitor>& visitor,
+                   ast::Program& ast) {
+    logger->info("Running {}", visitor.description);
+    visitor.v->visit_program(ast);
+    const std::string file = mod_file + "." + visitor.id + ".mod";
```

Comment:
nitpick: as above,

`const std::string file = "{}.{}.mod"_format(mod_file, visitor.id);`

Reply:
I like it! Here, performance matters less.